crates/bevy_ecs/src/component.rs

```diff
@@ -134,13 +134,15 @@ pub struct ComponentDescriptor {
     is_send_and_sync: bool,
     type_id: Option<TypeId>,
     layout: Layout,
-    drop: unsafe fn(*mut u8),
+    drop_multiple: unsafe fn(*mut u8, usize),
 }
 
 impl ComponentDescriptor {
-    // SAFETY: The pointer points to a valid value of type `T` and it is safe to drop this value.
-    unsafe fn drop_ptr<T>(x: *mut u8) {
-        x.cast::<T>().drop_in_place()
+    // SAFETY: The pointer points to a valid array of type `[T; len]` and it is safe to drop this value.
+    unsafe fn drop_ptr_multiple<T>(x: *mut u8, len: usize) {
+        for i in 0..len as isize {
+            x.cast::<T>().offset(i).drop_in_place()
+        }
```

Comment:
Couldn't this just be `std::slice::from_raw_parts_mut(x.cast::<T>(), len).drop_in_place()`?

Reply:
It did have to be the `std::ptr::` counterpart, but i think that could work.